dbus_test.go

```diff
@@ -0,0 +1,24 @@
+package dbus
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type TestStruct struct {
+	TestInt int
+	TestStr string
+}
+
+func Test_VariantOfStruct(t *testing.T) {
+	tester := TestStruct{TestInt: 123, TestStr: "foobar"}
+	testerDecoded := []interface{}{123, "foobar"}
+	variant := MakeVariant(testerDecoded)
+	input := []interface{}{variant}
+	output := new(TestStruct)
```

Comment:
The following refactor might read better and avoid the need for the additional library and the numerous dependencies:
```
var output TestStruct
if err := Store(input, &output); err != nil {
    t.Fatal(err)
}
if tester != output {
    t.Fatalf("%v != %v\n", tester, output)
}